modules/core/arrow-core-data/src/main/kotlin/arrow/unsafe.kt

```diff
@@ -1,18 +1,19 @@
 package arrow
 
 import arrow.typeclasses.Continuation
-import java.util.concurrent.atomic.AtomicReference
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
 import kotlin.coroutines.CoroutineContext
 import kotlin.coroutines.EmptyCoroutineContext
 import kotlin.coroutines.RestrictsSuspension
 import kotlin.coroutines.startCoroutine
 
 private class UnsafeContinuation<A>(
-  val result: AtomicReference<A> = AtomicReference()
+  val result: AtomicRef<A?> = atomic(null)
```

Comment:
Instead of nullable type can't we lateinit?

Reply:
I don't think so, if you take a closer look at Java's AtomicRefernece you'll see that they don't initialize the value, which means it will be null until initialized, and it will throw when we get it because is nullable and we expect A.

So this is the most straightforward conversion afaik unless we want to handle that case specifically